kodi_addon_checker/check_addon.py

```diff
@@ -302,3 +306,30 @@ def _check_file_whitelist(report: Report, file_index, addon_path):
 def relative_path(file_path):
     path_to_print = file_path[len(REL_PATH):]
     return ".{}".format(path_to_print)
+
+
+def _check_complex_addon_entrypoint(report: Report, addon_path, max_entrypoint_line_count):
+
+    addon_xml_path = os.path.join(addon_path, "addon.xml")
+    tree = xml.etree.ElementTree.parse(addon_xml_path).getroot()
+
+    for i in tree.findall("extension"):
+        library = i.get("library")
+
+        if library:
+            filepath = os.path.join(addon_path, library)
+
+            if not os.path.isdir(filepath):
+
+                if os.path.exists(filepath):
+                    if number_of_lines(filepath) > max_entrypoint_line_count:
+                        report.add(Record(WARNING, "Entry point for the addon is complex. please check %s" % library))
```

Comment:
This error message needs work. Would be helpful to mention how many lines we counted vs how many are allowed.
Maybe we should start to document warnings/problems in our wiki and link to those documentations in the errors?